packages/salesforcedx-vscode-core/src/deploys/toolingDeploy.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { AuthInfo, Connection } from '@salesforce/core';
+import * as fs from 'fs';
+import * as path from 'path';
+import {
+  supportedToolingTypes,
+  ToolingCreateResult,
+  ToolingRetrieveResult
+} from './index';
+// tslint:disable-next-line:no-var-requires
+const DOMParser = require('xmldom-sfdx-encoding').DOMParser;
+
+export class ToolingDeploy {
+  private static instance: ToolingDeploy;
+  private metadataType?: string;
+  private connection?: Connection;
```

Comment:
thinking about making these public in case users don't want to call the deploy() method and want more granular control over the operations